### PR TITLE
Project Permission & Default Permission Docs

### PR DESCRIPTION
**Pull request by martinbpeters**

### CODE REVIEW

**Comment by jacalata** [comment]
@dzucker-tab look good to you?

File: docs/api-ref.md [hunk 1]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ server.projects.delete_permission(project_item)
+``` 
+
+
+#### projects.populate_workbook_default_permissions
+```py
+projects.populate_workbook_default_permissions(item)
+```
+
+Queries the default workbook permissions, parses and stores the returned the permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:1755]: Take or leave these comments. I think what you have is legible. 

1. In descriptive content it is data source not datasource.

2. The name of the method already says this is about default permissions of a project, but the sentence, standalone, would include that context if it were:

Queries the default permissions of workbooks in a project, parses and stores the returned the permissions.

File: docs/api-ref.md [hunk 2]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ project_item = all_project_items[0]
+ server.projects.populate_workbook_default_permissions(project_item)
+``` 
+
+#### projects.populate_datasource_default_permissions
+```py
+projects.populate_datasource_default_permissions(item)
+```
+
+Queries the default datasource permissions, parses and stores the returned the permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:1782]: same as comment above, suggest:

Queries the default permissions of data sources in a project, parses and stores the returned the permissions.

File: docs/api-ref.md [hunk 3]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ server.projects.populate_datasource_default_permissions(project_item)
+``` 
+
+
+#### projects.populate_flow_default_permissions
+```py
+projects.populate_flow_default_permissions(item)
+```
+
+Queries the default flow permissions, parses and stores the returned the permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:1810]: Again:

Queries the default permissions for flows in a project, parses and stores the returned the permissions.

File: docs/api-ref.md [hunk 4]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ project_item = all_project_items[0]
+ server.projects.populate_flow_default_permissions(project_item)
+``` 
+
+#### projects.update_workbook_default_permissions
+```py
+projects.update_workbook_default_permissions(item, rules)
+```
+
+Add or update default workbook permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:1838]: Again:

Add or update default permissions for workbooks in a project.

File: docs/api-ref.md [hunk 5]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ 
+ server.projects.update_workbook_default_permissions(project_item, [rules])
+``` 
+
+#### projects.update_datasource_default_permissions
+```py
+projects.update_datasource_default_permissions(item, rules)
+```
+
+Add or update default datasource permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:1876]: You get the idea:

Add or update default permissions for data sources in a project.

File: docs/api-ref.md [hunk 6]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ server.projects.update_datasource_default_permissions(project_item, [rules])
+``` 
+
+
+#### projects.update_flow_default_permissions
+```py
+projects.update_flow_default_permissions(item, rules)
+```
+
+Add or update default flow permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:1915]: yawn . . . 

Add or update default permissions for flows in a project.

File: docs/api-ref.md [hunk 7]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ 
+ server.projects.update_flow_default_permissions(project_item, [rules])
+``` 
+
+#### projects.delete_workbook_default_permissions
+```py
+projects.delete_workbook_default_permissions(item, rules)
+```
+
+Delete default workbook permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:1953]: mm hm ...

Delete default permissions for workbooks in a project.

File: docs/api-ref.md [hunk 8]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ 
+ server.projects.delete_workbook_default_permissions(project_item, [rules])
+``` 
+
+#### projects.delete_datasource_default_permissions
+```py
+projects.delete_datasource_default_permissions(item, rules)
+```
+
+Delete default datasource permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:1991]: you guessed it:

Delete default permissions for data sources in a project.

File: docs/api-ref.md [hunk 9]
```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
...
+ 
+ server.projects.delete_datasource_default_permissions(project_item, [rules])
+``` 
+
+#### projects.delete_flow_default_permissions
+```py
+projects.delete_flow_default_permissions(item, rules)
+```
+
+Delete default flow permissions.
```
Comment (dzucker-tab) [inline docs/api-ref.md:2029]: that's right:

Delete default permissions for flows in a project.

**Pull request merged**